csharp/src/Google.Protobuf/ByteString.cs

```diff
@@ -140,7 +140,7 @@ public static ByteString FromBase64(string bytes)
         {
             // By handling the empty string explicitly, we not only optimize but we fix a
             // problem on CF 2.0. See issue 61 for details.
-            return bytes == "" ? Empty : new ByteString(Convert.FromBase64String(bytes));
+            return bytes.Length == 0 ? Empty : new ByteString(Convert.FromBase64String(bytes));
```

Comment:
This will change from ArgumentNullException (thrown by Convert.FromBase64String) to NullReferenceException if `bytes` is null.